src/operations/blas1_trees.hpp

```diff
@@ -619,6 +619,275 @@ Rotg<operand_t>::adjust_access_displacement() {
   s_.adjust_access_displacement();
 }
 
+template <typename operand_t>
+Rotmg<operand_t>::Rotmg(operand_t &_d1, operand_t &_d2, operand_t &_x1,
+                        operand_t &_y1, operand_t &_param)
+    : d1_{_d1}, d2_{_d2}, x1_{_x1}, y1_{_y1}, param_{_param} {}
+
+template <typename operand_t>
+SYCL_BLAS_INLINE typename Rotmg<operand_t>::index_t Rotmg<operand_t>::get_size()
+    const {
+  return static_cast<Rotmg<operand_t>::index_t>(1);
+}
+
+/**
+ * For further details about the rotmg algorithm refer to:
+ *
+ * Hopkins, Tim (1998) Restructuring the BLAS Level 1 Routine for Computing the
+ * Modified Givens Transformation. Technical report. , Canterbury, Kent, UK.
+ *
+ * and
+ *
+ * C. L. Lawson, R. J. Hanson, D. R. Kincaid, and F. T. Krogh. Basic linear algebra
+ * subprograms for Fortran usage. ACM Trans. Math. Softw., 5:308-323, 1979.
+ */
+template<typename operand_t>
+SYCL_BLAS_INLINE typename Rotmg<operand_t>::value_t Rotmg<operand_t>::eval(
+    typename Rotmg<operand_t>::index_t i) {
+
+  using zero = constant<value_t, const_val::zero>;
+  using one = constant<value_t, const_val::one>;
+  using two = constant<value_t, const_val::two>;
+  using m_one = constant<value_t, const_val::m_one>;
+  using m_two = constant<value_t, const_val::m_two>;
+
+  using error = two;
+
+  using clts_flag = one; /* co-sin less than sin */
+  using sltc_flag = zero; /* sin less than co-sin */
+  using rescaled_flag = m_one;
+  using unit_flag = m_two;
+
+  /* Gamma is a magic number used to re-scale the output and avoid underflows or
+   * overflows. Consult the papers above for more info */
+  constexpr value_t gamma = static_cast<value_t>(4096.0);
+
+  /* Square of gamma. It is hardcoded to avoid computing it on every call */
+  constexpr value_t gamma_sq = gamma * gamma;
+
+  /* Inverse of the square of gamma (i.e. 1 / (gamma * gamma)) */
+  constexpr value_t inv_gamma_sq = static_cast<value_t>(1.0) / gamma;
```

Comment:
Do these ever get used? The netlib lapack uses 1 / (gamma * gamma), instead of 1 * inv_gamma_sq too.  I'd be interested if these gave different results!

Reply:
Good catch. This is a bug that I introduced when moving from magic numbers to variables. And somehow justified to myself with cblas implementation being different.